File: Ecommerce.Core/Models/Entities/Category.cs
using System;
using System.Collections.Generic;

namespace Ecommerce.Core.Models.Entities
{
    public class Category : BaseEntity
    {
        public string CategoryName { get; set; }
        public string CategoryDescription { get; set; }
        public string CategorySlug { get; set; }
        public string CategoryImageUrl { get; set; }
        public int DisplayOrder { get; set; }
        public bool IsActive { get; set; } = true;
        public virtual ICollection<Product> Products { get; set; } = new List<Product>();
    }
} 

File: Ecommerce.Admin/src/models/Category.ts
export interface Category {
  id: string;
  categoryName: string;
  categoryDescription?: string;
  categorySlug?: string;
  categoryImageUrl?: string;
  displayOrder: number;
  isActive: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateCategoryDto {
  categoryName: string;
  categoryDescription?: string;
  categorySlug?: string;
  categoryImageUrl?: string;
  categoryImage?: File;
  displayOrder: number;
  isActive: boolean;
}

export interface UpdateCategoryDto extends CreateCategoryDto {
  id: string;
} 